### PR TITLE
add-feedback-email-functionality

### PR DESCRIPTION
**Pull request by davidthomas22**
### Context

> Does this issue have a Trello card?

https://trello.com/c/3Nt7wFgJ

### Intent

> What changes are introduced by this PR that correspond to the above card?

What
Enable emailed feedback to go to each DCM once a week

Why
So that we can share the existing feedback without having to give access to data studio

### Checklist

- [ ] This PR contains **only** changes related to the above card
- [ ] Tests have been added/updated to cover the change
- [ ] Documentation has been updated where appropriate
- [ ] Tested in Development

### CODE REVIEW

File: job/config.ts [hunk 1]
```diff
@@ -14,4 +16,8 @@ export = {
     serviceAccountKey,
     spreadsheetId,
   },
+  govNotify: {
```
Comment (davidthomas22) [inline job/config.ts:19]: config for govNotify

File: job/feedbackEmailSender.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+// import logger from './utils/logger'
+import { format } from 'date-fns'
+import type { FeedbackItem } from './feedbackRetriever'
+import config from './config'
+
+type Contact = { name: string; establishment: string; email: string }
+
+const establishmentFeedback = (feedbackItems: FeedbackItem[]) => {
```
Comment (davidthomas22) [inline job/feedbackEmailSender.ts:8]: Function to create new array of feedback items by establishment

File: job/feedbackEmailSender.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    if (establishmentSpecificFeedback.has(establishment)) {
+      const matchingEstablishment = establishmentSpecificFeedback.get(establishment)
+      matchingEstablishment.push(feedbackItem)
+      establishmentSpecificFeedback.set(establishment, matchingEstablishment)
+    } else establishmentSpecificFeedback.set(establishment, [feedbackItem])
+  })
+  return establishmentSpecificFeedback
+}
+
+const establishmentStaff = (contactList: Contact[]) => {
```
Comment (davidthomas22) [inline job/feedbackEmailSender.ts:22]: Function to create new array of contacts by establishment

File: job/feedbackEmailSender.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    const { establishment } = contact
+    if (establishmentContactList.has(establishment)) {
+      const matchingEstablishment = establishmentContactList.get(establishment)
+      matchingEstablishment.push(contact)
+      establishmentContactList.set(establishment, matchingEstablishment)
+    } else establishmentContactList.set(establishment, [contact])
+  })
+  return establishmentContactList
+}
+class FeedbackEmailSender {
```
Comment (davidthomas22) [inline job/feedbackEmailSender.ts:35]: Function to combine feedback items and contacts by establishment / post and send email via Gov notify

File: job/__tests__/feedbackJob.spec.ts [hunk 5]
```diff
@@ -25,10 +30,18 @@ describe('feedback retriever', () => {
       expect(feedbackRetriever.retrieve).toHaveBeenCalledWith('2021-02-21', '2021-02-21')
     })
 
-    it('should pass the results of the retriever to the upload', async () => {
-      feedbackRetriever.retrieve.mockResolvedValue([['cell1']])
+    it('should pass the results of the retriever to the sheets uploader', async () => {
+      const feedbackItem = new FeedbackItem({})
+      feedbackRetriever.retrieve.mockResolvedValue([feedbackItem])
+      await feedbackJob.run(new Date('22 Feb 2021 00:00:00 GMT'))
```
Comment (jonhindle) [inline job/__tests__/feedbackJob.spec.ts:36]: Duplicated twice. Could have been in a beforeEach block.
```
beforeEach(() => {
    const feedbackItem = new FeedbackItem({})
    feedbackRetriever.retrieve.mockResolvedValue([feedbackItem])
    await feedbackJob.run(new Date('22 Feb 2021 00:00:00 GMT'))
})
```

**Review by jonhindle — DISMISSED** [review]
lgtm

File: README.md [hunk 6]
```diff
@@ -22,4 +12,32 @@ To compile and run the job run:
...
+  SPREADSHEET_ID=some-value
+  GOV_NOTIFY_API_KEY=some-value
+```
+
+also to test the gov notify email functionality locally, ensure the following env var is set:
+```
+  NOTIFICATION_DAY=current-day-value
+```
+
+(Establishment must match the (uppercase) drupal id for the establishment)
```
Comment (andrewrlee) [inline README.md:35]: guess this line can go now?

Comment (davidthomas22) [reply README.md:35]: Agreed

File: job/types.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+      this.establishment,
+      this.series,
+    ]
+  }
+
+  formattedRow() {
+    return this.row
+      .map(cell => `"${cell}"`)
+      .join(',')
+      .replaceAll(undefined, '')
```
[the remark this answers is not in the document]
Comment (davidthomas22) [reply job/types.ts:39]: Of course, yes

**Review by andrewrlee — DISMISSED** [review]
LGTM

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**